keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClientBuilder.java

```diff
@@ -0,0 +1,203 @@
+     * {@link KeyAsyncClientBuilder#endpoint(String) serviceEndpoint} are used to create the
+     * {@link KeyAsyncClientBuilder client}. All other builder settings are ignored. If {@code pipeline} is not set,
+     * then {@link KeyAsyncClientBuilder#credentials(TokenCredential) key vault credentials and
+     * {@link KeyAsyncClientBuilder#endpoint(String)} key vault endpoint are required to build the {@link KeyAsyncClient client}.}</p>
+     *
+     * @return A KeyAsyncClient with the options set from the builder.
+     * @throws IllegalStateException If {@link KeyAsyncClientBuilder#credentials(TokenCredential)} or
+     * {@link KeyAsyncClientBuilder#endpoint(String)} have not been set.
+     */
+    public KeyAsyncClient build() {
+
+        if (endpoint == null) {
+            throw new IllegalStateException(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.VAULT_END_POINT_REQUIRED));
+        }
+
+        if (pipeline != null) {
+            return new KeyAsyncClient(endpoint, pipeline);
+        }
+
+        if (credentials == null) {
+            throw new IllegalStateException(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.CREDENTIALS_REQUIRED));
+        }
+
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+        policies.add(new UserAgentPolicy(AzureKeyVaultConfiguration.SDK_NAME, AzureKeyVaultConfiguration.SDK_VERSION));
+        policies.add(retryPolicy);
+        policies.add(new TokenCredentialPolicy(credentials));
+        policies.addAll(this.policies);
+        policies.add(new HttpLoggingPolicy(httpLogDetailLevel));
+
+        HttpPipeline pipeline = HttpPipeline.builder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build();
+
+        return new KeyAsyncClient(endpoint, pipeline);
+    }
+
+    /**
+     * Sets the vault endpoint url to send HTTP requests to.
+     *
+     * @param endPoint The vault endpoint url is used as destination on Azure to send requests to.
+     * @return the updated Builder object.
+     * @throws IllegalStateException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public KeyAsyncClientBuilder endpoint(String endPoint) {
+        try {
+            this.endpoint = new URL(endPoint);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("The Azure Key Vault endpoint url is malformed.");
+        }
+        return this;
+    }
+
+    /**
+     * Sets the credentials to use when authenticating HTTP requests.
+     *
+     * @param credentials The credentials to use for authenticating HTTP requests.
+     * @return the updated Builder object.
+     * @throws NullPointerException if {@code credentials} is {@code null}.
+     */
+    public KeyAsyncClientBuilder credentials(TokenCredential credentials) {
```

Comment:
credential should be singular throughout the library.

Reply:
updated to credential singular everywhere.